src/Gating/IGate.cs

```diff
@@ -147,6 +147,12 @@ public interface IGate
 		DateTime? EndDate { get; }
 
 
+		/// <summary>
+		/// Set of cloud contexts that apply to this gate
+		/// </summary>
+		HashSet<string> CloudContexts { get; }
```

Comment:
>HashSet [](start = 2, length = 7)

Suggestion: as this is generic interface, ISet<string> would be more appropriate